var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -228,54 +228,55 @@ def install(self, spec, prefix):
 
         # take step-40 which can use both PETSc and Trilinos
         # FIXME: switch step-40 to MPI run
-        with working_dir('examples/step-40'):
-            print('=====================================')
-            print('========== Step-40 PETSc ============')
-            print('=====================================')
-            # list the number of cycles to speed up
-            filter_file(r'(const unsigned int n_cycles = 8;)',
-                        ('const unsigned int n_cycles = 2;'), 'step-40.cc')
-            cmake('.')
-            if '^petsc' in spec:
-                make('release')
-                make('run', parallel=False)
-
-            print('=====================================')
-            print('========= Step-40 Trilinos ==========')
-            print('=====================================')
-            # change Linear Algebra to Trilinos
-            filter_file(r'(\/\/ #define FORCE_USE_OF_TRILINOS.*)',
-                        ('#define FORCE_USE_OF_TRILINOS'), 'step-40.cc')
-            if '^trilinos+hypre' in spec:
-                make('release')
-                make('run', parallel=False)
+        if spec.satisfies('@8.4.0:'):
+            with working_dir('examples/step-40'):
+                print('=====================================')
+                print('========== Step-40 PETSc ============')
+                print('=====================================')
+                # list the number of cycles to speed up
+                filter_file(r'(const unsigned int n_cycles = 8;)',
+                            ('const unsigned int n_cycles = 2;'), 'step-40.cc')
+                cmake('.')
+                if '^petsc' in spec:
+                    make('release')
+                    make('run', parallel=False)
+            
+                print('=====================================')
+                print('========= Step-40 Trilinos ==========')
+                print('=====================================')
+                # change Linear Algebra to Trilinos
+                filter_file(r'(\/\/ #define FORCE_USE_OF_TRILINOS.*)',
+                            ('#define FORCE_USE_OF_TRILINOS'), 'step-40.cc')
+                if '^trilinos+hypre' in spec:
+                    make('release')
+                    make('run', parallel=False)
 
-            print('=====================================')
-            print('=== Step-40 Trilinos SuperluDist ====')
-            print('=====================================')
-            # change to direct solvers
-            filter_file(r'(LA::SolverCG solver\(solver_control\);)',  ('TrilinosWrappers::SolverDirect::AdditionalData data(false,"Amesos_Superludist"); TrilinosWrappers::SolverDirect solver(solver_control,data);'), 'step-40.cc')  # NOQA: ignore=E501
```

Comment:
I think I know the problem. 
Does "filter_file" change the actual file or does it use a buffer or temp file? 
Since I first installed it without checking the version, "filter_file" has tried to change it and removed the lines (I am testing this right now with a clean download and recompile which takes some time).

Another thing is that if "filter_file" changes the actual file then I don't know if it is a good idea to do the tests like this. Since the examples are changed from what is shown in dealii website.